### PR TITLE
Health Regen

### PR DESCRIPTION
**Pull request by AntonAsbrink**
Gives the player passive health regeneration upgrade. This took unresonably long to make because of a bug with player upgrades where they were not upgraded on the first level because...

### CODE REVIEW

File: StortSpelProjekt/Game/src/Gamefiles/Components/UpgradeComponents/Upgrades/UpgradeHealthRegen.cpp [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+	// increase the price of the upgrade
+	m_Price += m_StartingPrice;
+}
+
+std::string UpgradeHealthRegen::GetDescription(unsigned int level)
+{
+	std::ostringstream str;
+	str << std::fixed << std::setprecision(2) << (3.0f / level);
+	return "Health Regenration: Passively heals the player 1 health point every " + str.str();
```
Comment (jocke1995) [inline StortSpelProjekt/Game/src/Gamefiles/Components/UpgradeComponents/Upgrades/UpgradeHealthRegen.cpp]: Spelling of "Regeneration" is wrong.

File: StortSpelProjekt/Game/src/Gamefiles/Components/UpgradeComponents/Upgrades/UpgradeHealthRegen.h [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+	void IncreaseLevel();
+
+	std::string GetDescription(unsigned int level);
+
+private:
+	double m_HealthTimer;
+
+};
+
+#endif // !UPGRADE_HEALTH_BOOST_H
```
Comment (jocke1995) [inline StortSpelProjekt/Game/src/Gamefiles/Components/UpgradeComponents/Upgrades/UpgradeHealthRegen.h]: Comment should be HEALTH_REGEN right?

**Review by jocke1995 — APPROVED** [review]
Looks good

**Review by jocke1995 — CHANGES_REQUESTED** [review]
some small comments

**Review by jocke1995 — APPROVED** [review]

**Review by LeoWikstrom — APPROVED** [review]
Love the icon ❤️ 
100 seems fair

**Pull request merged**